documentation/kernel-semantics.tex

```diff
@@ -3174,14 +3214,113 @@ \subsubsection{Async}
+                {\text{where } \econt' = \AwaitEK}
+        \end{multlined}\label{async:await}\\
+        &\begin{multlined}
+            \cesktranswheresplit%
+                {\evalconf{\StaticInvocation{\{\membermeta\}}{\expressionsmeta}}{\env}{\strace}{\cstrace}{\cex}{\econt}}%
+                {\evallistconf{\exprs}{\env}{\strace}{\cstrace}{\cex}{\acont}}%
+                {\begin{aligned}
+                    \text{where }
+                    \acont &= \AsyncStaticInvA{\formals}{\stmt}{\strace'}{\econt},\\
+                    \strace' &= \StaticInvocation{\{\membermeta\}}{\expressionsmeta}::\strace,\\
+                    \membermeta &\text{ is an async static method with formal parameters }\formals \text{ and body }\stmt
+                  \end{aligned}}
+        \end{multlined}\label{async:static-invoc}\\
+        &\begin{multlined}
+             \cesktranswheresplit%
+                 {\acontconf{\AsyncStaticInvA{\formals}{\stmt}{\strace}{\econt}}{\val{s}}}%
+                 {\contconf{\econt}{\val}}%
+                 {\begin{aligned}
+                     \text{where }
+                     \val &= FutureValue(\loc)\\
+                     \loc &= \text{fresh location in the store}\\
+                     \eventloop_{R} &= \append(\eventloop_{L},\ncont)\\
+                     \ncont &= \StartNK{\stmt}{\env}{\strace}{\econt'}\\
+                     \econt' &= \CompleteFutureEK{\loc}\\
+                     \env &= \ext{\env_{empty}}{\formals}{\val{s}}
+                  \end{aligned}}
+        \end{multlined}\label{async:static-invoc-a}\\
+        &\begin{multlined}
+             \cesktranswheresplit%
+                 {\contconf{\AwaitEK}{\val}}%
+                 {\eventconf{\head(\eventloop)}{\eventloop'}}%
+                 {\begin{aligned}
+                     \text{where }
+                     \val &\in \textbf{FutureValue}\\
+                     \eventloop' &= \tail(\append(\eventloop, \AwaitReturnNK{\econt}{\loc})
+                 \end{aligned}}
+        \end{multlined}\label{async:econt-future}\\
+        &\begin{multlined}
+             \cesktranswhere%
+                 {\contconf{\AwaitEK}{\val}}%
+                 {\contconf{\econt}{\val}}%
+                 {\text{where }\val \notin \textbf{FutureValue}}
+        \end{multlined}\label{async:econt-value}\\
+        &\begin{multlined}
+             \cesktranswheresplit%
+                 {\eventconf{\StartNK{\stmt}{\env}{\strace}{\econt}}{\eventloop}}%
+                 {\execconf{\stmt}{\env}{[]}{[]}{\strace}{\handler}{\cex,\,\cstrace}{\econt}{\scont}}%
+                 {\text{where } \scont = \ExitSK{null}}
+        \end{multlined}\label{async:startnk}\\
+         &\begin{multlined}
+             \cesktranswheresplit%
+                 {\contconf{\CompleteFutureEK{\loc}}{\val}}%
+                 {\eventconf{\head(G)}{\tail(G)}}%
+                 {\text{where } \deref{\loc} = \val \text{ after transition}}
```

Comment:
I assumed any of  `update`, or "after transition" wording may be used to unify after our discussion. I had previously updated the text to use "after transition" wording, and just followed this convention here. I may update this with another change to avoid merge issues with other PRs.